src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -321,5 +323,11 @@ internal static DependencyGraphSpec CreateMinimalDependencyGraphSpec(string proj
 
             return dgSpec;
         }
+
+        public Task RegisterRestoreInfoSourceAsync(IVsProjectRestoreInfoSource restoreInfoSource, CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
```

Comment:
Just one question.
When we use `throw not ImplementedException` vs `Task.CompletedTask` ?

Reply:
This is completely temporary. I don't want the tests on project-system end to throw exceptions. It's up to NuGet to use the project-system in this case.